### PR TITLE
Fix ineffectual assignments

### PR DESCRIPTION
**Pull request by yuva29**
Fix errors reported by ineffassign 

Signed-off-by: Yuva Shankar <yuva29@users.noreply.github.com>

### CODE REVIEW

File: test/systemtests/k8setup_test.go [hunk 1]
```diff
@@ -364,7 +364,7 @@ func (k *kubernetes) tcFilterShow(bw string) error {
 	rate := strings.Split(output[1], "burst")
 	regex := regexp.MustCompile("[0-9]+")
 	outputStr := regex.FindAllString(rate[0], -1)
-	outputInt, err := strconv.ParseInt(outputStr[0], 10, 64)
+	outputInt, _ := strconv.ParseInt(outputStr[0], 10, 64)
```
Comment (dseevr) [inline test/systemtests/k8setup_test.go]: Same here

File: test/systemtests/swarm_test.go [hunk 2]
```diff
@@ -336,7 +337,7 @@ func (w *swarm) tcFilterShow(bw string) error {
 	rate := strings.Split(output[1], "burst")
 	regex := regexp.MustCompile("[0-9]+")
 	outputStr := regex.FindAllString(rate[0], -1)
-	outputInt, err := strconv.ParseInt(outputStr[0], 10, 64)
+	outputInt, _ := strconv.ParseInt(outputStr[0], 10, 64)
```
Comment (dseevr) [inline test/systemtests/swarm_test.go]: Same here

File: test/integration/utils_test.go [hunk 3]
```diff
@@ -347,7 +347,7 @@ func tcFilterCheckBw(expBw, expBurst int64) error {
 	rate := strings.Split(output[1], "burst")
 	regex := regexp.MustCompile("[0-9]+")
 	outputStr := regex.FindAllString(rate[0], -1)
-	outputInt, err := strconv.ParseInt(outputStr[0], 10, 64)
+	outputInt, _ := strconv.ParseInt(outputStr[0], 10, 64)
```
Comment (dseevr) [inline test/integration/utils_test.go]: I would just omit the second assignment here like you did in docker_test.go

File: netctl/netctl.go [hunk 4]
```diff
@@ -576,6 +576,10 @@ func inspectTenant(ctx *cli.Context) {
 	errCheck(ctx, err)
 
 	content, err := json.MarshalIndent(ten, "", "  ")
+	if err != nil {
+		log.Debugf("Failed to marshal indent %v", ten)
```
Comment (dseevr) [inline netctl/netctl.go]: You need to import logrus at the top here

**Comment by unclejack** [comment]
@yuva29: The CI has failed. It seems to be related to the code changes made by the PR.

**Comment by yuva29** [comment]
@unclejack Yeah. I'm debugging it. I'll push it again once i'm done.

File: netplugin/plugin/netplugin_test.go [hunk 5]
```diff
@@ -195,28 +198,3 @@ func TestNetPluginInitInvalidConfigMissingNetworkDriverName(t *testing.T) {
 		t.Fatalf("plugin init succeeded, should have failed!")
 	}
 }
-
-func TestNetPluginInitInvalidConfigMissingFwdMode(t *testing.T) {
```
Comment (yuva29) [inline netplugin/plugin/netplugin_test.go]: Removing this test as its no longer valid. https://github.com/contiv/netplugin/blob/0f843fa92f923f6f37378c3956ae02ab3855350e/netplugin/plugin/netplugin.go#L299 there's always a fwd mode.

File: drivers/ovsdriver.go [hunk 6]
```diff
@@ -843,6 +843,10 @@ func (d *OvsDriver) InspectNameserver() ([]byte, error) {
 	}
 
 	ns, err := d.nameServer.InspectState()
+	if err != nil {
+		log.Errorf("Error inspecting nameserver state: %v", err)
```
Comment (rchirakk) [inline drivers/ovsdriver.go]: 'return' in case of error

File: drivers/ovsdbDriver.go [hunk 7]
```diff
@@ -217,7 +217,7 @@ func (d *OvsdbDriver) createDeleteBridge(bridgeName, failMode string, op oper) e
 	}
 
 	// simple insert/delete operation
-	brOp := libovsdb.Operation{}
+	var brOp libovsdb.Operation
```
Comment (jojimt) [inline drivers/ovsdbDriver.go]: If I understood the golang right, the above two statements have the same effect. Was this flagged by the tool? Just curious.

**Comment by jojimt** [comment]
"Fix errors reported by ineffassign" @yuva29 is there a reference doc for this tool somewhere?

**Comment by unclejack** [comment]
@jojimt: I don't know if there's a reference. However, you can install the tool using `go get github.com/gordonklaus/ineffassign` (I've mentioned this because some tools have a sub-package called cmd where the actual source for the installable binary lives) and take a look at the repository here: https://github.com/gordonklaus/ineffassign.

**Comment by jojimt** [comment]
The reason I ask is some of the changes look questionable to me. It is easy for personal taste to creep in disguised as a tool and I would like to avoid that. E.g. many of the patterns addressed here have been seen extensively in other open source repos. I am not saying no here. But I would like to understand a bit better as to what we are doing here.

**Comment by yuva29** [comment]
@jojimt yeah right. These are errors flagged by https://github.com/gordonklaus/ineffassign. Its the same ineffassign used in go report card here https://goreportcard.com/report/github.com/contiv/netplugin#ineffassign

**Comment by dseevr** [comment]
FWIW, the actual Go source tends to prefer the `var`-style declaration when a variable should have its default value and when a struct is being initialized with no custom values

**Comment by rchirakk** [comment]
I think the tools is NOT checking  'var' vs short declaration.
It checks if the initial value is being used or not.
for example
```for i:=0; i<10; i++ {}```
is not an flagged  as initial value i=0 is used inside the for loop
same for 
```
j:=0
for ;; j<10 {j++}
```

**Comment by dseevr** [comment]
@yuva29 could you amend this PR so that `ineffassign` runs automatically like the other tools?  I think the `misspell` PR I added was the most recent one which you could use as a reference (use blame to find the commit if you need it)

**Pull request closed without merging**